src/tree/mod.rs

```diff
@@ -1134,6 +1248,10 @@ impl<T> Tree<T> {
     }
 }
 
+trait IteratorNew<'a, T, I> {
+    fn new(tree: &'a Tree<T>, node_id: NodeId) -> I;
+}
+
```

Comment:
Because the Iterators are in a lower module, I think you would not need to make them public.
But of course, if this is about visibility, it is understandable.